decidim-surveys/app/serializers/decidim/surveys/data_importer.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Surveys
+    # Importer for Surveys specific data (this is, its questionnaire).
+    class DataImporter < Decidim::Importers::Importer
+      def initialize(component)
+        @component = component
+      end
+
+      # Public: Creates a new Decidim::Surveys::Survey and Decidim::Forms::Questionnaire associated to the given +component+
+      #         for each serialized survey object.
+      # It imports the whole tree of Survey->Questionnaire->questions->answer_options.
+      #
+      # serialized        - The Hash of attributes for the Questionnaire and its relations.
+      # user              - The +user+ that is performing this action
+      #
+      # Returns the ser.
+      def import(serialized, user)
+        # we duplicate so that we can delete without affecting the received Hash
+        serialized.dup.collect do |serialized_survey|
```

Comment:
I think this should be wrapped in a transaction, otherwise an error could cause to have orphan or incomplete data.

Reply:
It is already in a transaction, because its called from Decidim::ParticipatoryProcesses::Admin::ImportParticipatoryProcess:L27.
Do you mean to have it inside a sub-transaction in the importer itself? I'll do it, it won't cause any harm.